x-pack/plugin/vectors/src/test/java/org/elasticsearch/xpack/vectors/mapper/VectorEncoderDecoderTests.java

```diff
@@ -16,15 +17,39 @@
 public class VectorEncoderDecoderTests extends ESTestCase {
 
     public void testDenseVectorEncodingDecoding() {
+        Version indexVersion = Version.CURRENT;
         int dimCount = randomIntBetween(0, DenseVectorFieldMapper.MAX_DIMS_COUNT);
         float[] expectedValues = new float[dimCount];
+        double dotProduct = 0f;
         for (int i = 0; i < dimCount; i++) {
             expectedValues[i] = randomFloat();
+            dotProduct += expectedValues[i] * expectedValues[i];
         }
+        float expectedMagnitude = (float) Math.sqrt(dotProduct);
 
         // test that values that went through encoding and decoding are equal to their original
         BytesRef encodedDenseVector = mockEncodeDenseVector(expectedValues);
-        float[] decodedValues = VectorEncoderDecoder.decodeDenseVector(encodedDenseVector);
+        float[] decodedValues = VectorEncoderDecoder.decodeDenseVector(indexVersion, encodedDenseVector);
+        float decodedMagnitude = VectorEncoderDecoder.getVectorMagnitude(indexVersion, encodedDenseVector, decodedValues);
+        assertEquals(expectedMagnitude, decodedMagnitude, 0.001f);
```

Comment:
Should this error threshold be `0.0f`, since we don't expect any loss of precision from encoding and decoding?